src/components/map/HfpMarkerLayer.js

```diff
@@ -65,16 +65,8 @@ class HfpMarkerLayer extends Component {
   };
 
   onMarkerClick = (positionWhenClicked) => () => {
-    const {onMarkerClick, selectedJourney} = this.props;
-
-    onMarkerClick(
-      get(selectedJourney, "uniqueVehicleId", "") !==
-        positionWhenClicked.uniqueVehicleId ||
-      get(selectedJourney, "journeyStartTime", "") !==
-        positionWhenClicked.journeyStartTime
-        ? positionWhenClicked
-        : null
-    );
+    const {onMarkerClick} = this.props;
+    onMarkerClick(positionWhenClicked);
```

Comment:
The setJourney action nulls the selectedJourney if trying to select it (it acts as a toggle), so we don't need to do that here.